qiskit_machine_learning/neural_networks/sampling_qnn.py

```diff
@@ -0,0 +1,177 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A Sampling Neural Network implementation based on the Sampler primitive."""
+
+from __future__ import annotations
+
+from typing import Sequence, Callable
+
+import numpy as np
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.primitives import BaseSampler, SamplerResult
+from qiskit.utils import algorithm_globals
+
+from .sampler_qnn import SamplerQNN
+
+
+class SamplingQNN(SamplerQNN):
```

Comment:
How about having this feature in `SamplerQNN` like `CircuitQNN` using `sampling: bool` option?